easyfinance.client/src/app/features/project/detail-project/detail-project.component.ts

```diff
@@ -45,8 +48,9 @@ export class DetailProjectComponent implements OnInit {
   year: { budget: number, spend: number, overspend: number, remaining: number, earned: number; } = { budget: 0, spend: 0, overspend: 0, remaining: 0, earned: 0 };
   buttons: string[] = [this.btnIncome, this.btnCategory];
   showCopyPreviousButton = false;
+  transactions: { date: Date, description: string, amount: number, type: string }[] = [];
```

Comment:
```suggestion
  private transactions: BehaviorSubject<TransactionDto[]> = new BehaviorSubject<TransactionDto[]>([new TransactionDto()]);
  transactions$: Observable<TransactionDto[]> = this.transactions.asObservable();
```

I suggest you to use observable and create a `TransactionDto` to use here. Keeping the pattern as in the others component, you can check them if you need help.